### PR TITLE
[1LP][RFR] Implement regex search

### PR DESCRIPTION
**Pull request by sasoc**
With this commit we implement search with regular expression that uses Merkyl plugin.

Here is a simple use case that searches `evm.log` for `ems_ref` that starts with `1` and ends with `9`:
```python
import re

def test(merkyl_inspector):
    log_path = '/var/www/miq/vmdb/log/evm.log'
    merkyl_inspector.add(log_path) # <- add log that you want to track with merkyl
    simple_regex = re.compile(r'^.*ems_ref=1[0-9]+9 .*$', re.MULTILINE) # <- compile regex

    if merkyl_inspector.search_log_with_regex(simple_regex, log_path): # <- search for regex in evm.log
        print('ems_ref was found')
```

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 21a362d0c0341e0e196888c9f7e6032245540b80:

`cfme/fixtures/merkyl.py`:
- :bangbang: Line 72:58: **E251** *unexpected spaces around keyword / parameter equals*
- :bangbang: Line 72:60: **E251** *unexpected spaces around keyword / parameter equals*
- :bangbang: Line 98:1: **E303** *too many blank lines (4)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by sasoc** [comment]
cc @miha-plesko

**Comment by sasoc** [comment]
@miha-plesko Thank you for your rewiev, I've commited changes that you requested. I've also changed that `search_log_with_regex` now gets compiled regex. Would you please take another look.

File: cfme/fixtures/merkyl.py [hunk 1]
```diff
@@ -1,5 +1,43 @@
+"""
+To use merkyl you have to configure artifactor in conf/env.local.yaml
+and you have to open port 8291 for you appliance.
+
+    example config:
```
[the remark this answers is not in the document]
Comment (sasoc) [reply cfme/fixtures/merkyl.py]: You are right it really looks like part of the configuration, I've changed it.

**Comment by sasoc** [comment]
@miha-plesko I can move it all into `search_log`. It gets less obvious that you can use regular expression, but with right comment it can be. So do I put it all in search_log?

**Comment by miha-plesko** [comment]
Yes please. I understand your concern but TBH my first assumption when I've seen "search_log" name was that it searches by regex. As you say, a comment will clear things up 👍

**Comment by sasoc** [comment]
@miha-plesko I've commited changes, now its all implemented in `search_log`. Would you please check if its ok.

**Comment by miha-plesko** [comment]
@sasoc please set RFR prefix, I think we're good to go

**Comment by jawatts** [comment]
Moving to 1LP since @RonnyPfannschmidt  approved

File: cfme/fixtures/merkyl.py [hunk 2]
```diff
@@ -1,5 +1,44 @@
+"""
+To use merkyl you have to configure artifactor in conf/env.local.yaml
```
Comment (mshriver) [inline cfme/fixtures/merkyl.py]: :tada:

**Pull request merged**